system/lib/wasmfs/wasmfs.cpp

```diff
@@ -76,19 +77,80 @@ __wasi_errno_t __wasi_fd_write(__wasi_fd_t fd,
 
   auto lockedFile = file->locked();
 
-  __wasi_size_t num = 0;
+  off_t offset = lockedOpenFile.position();
   for (size_t i = 0; i < iovs_len; i++) {
     const uint8_t* buf = iovs[i].buf;
-    __wasi_size_t len = iovs[i].buf_len;
+    size_t len = iovs[i].buf_len;
 
-    lockedFile.write(buf, len);
-    num += len;
+    // Check if the sum of the buf_len values overflows an ssize_t value.
+    if (offset + len < offset) {
```

Comment:
It looks like `off_t` is defined to be `_Int64`, i.e. `int64_t`. Since that's a signed type, overflow on addition is undefined behavior, so to avoid UB it would be good to cast the relevant values to unsigned types when doing this check.